README.md

```diff
@@ -1,15 +1,91 @@
+<a href="https://unidist.readthedocs.io/en/latest/?badge=latest"><img alt="" src="https://readthedocs.org/projects/unidist/badge/?version=latest" align="center"></a>
+</p>
 
-To run tests you need to install ``pytest`` package:
+### What is unidist?
 
+unidist is a framework that is intended to provide the unified API for distributed execution by supporting various performant execution backends and utilizing all of your CPUs cores. At the moment the following backends
+are supported underneath the hood:
+
+* [Ray](https://docs.ray.io/en/master/index.html)
+* [MPI](https://www.mpi-forum.org/)
+* [Dask Distributed](https://distributed.dask.org/en/latest/)
+* [Python Multiprocessing](https://docs.python.org/3/library/multiprocessing.html)
+
+unidist is designed to work in a [task-based parallel](https://en.wikipedia.org/wiki/Task_parallelism) model.
+
+Also, the framework provides a sequential ``Python`` backend, that can be used for debugging.
+
+### Instalation
+
+Currently, unidist can be installed from sources using ``pip``:
+
+```bash
+$ pip install git+https://github.com/modin-project/unidist
 ```
-pip install pytest
+
+This will install unidist directly from the repo without you having to manually clone it! Please be aware that these changes have not made it into a release and may not be completely stable.
+
+#### Choosing an execution backend
+
+There are several ways to choose the execution backend for distributing computations.
+First, the recommended way is to use
+[unidist CLI](https://unidist.readthedocs.io/en/latest/using_cli.html) options:
+
+```bash
+# Running the script with unidist on Ray backend
+$ unidist script.py --backend ray
+# Running the script with unidist on MPI backend
+$ unidist script.py --backend mpi
 ```
-Run tests using command:
+
+Second, setting the environment variable:
+
+```bash
+# unidist will use Ray backend to distribute computations
+export UNIDIST_BACKEND=ray
+# unidist will use MPI backend to distribute computations
+export UNIDIST_BACKEND=mpi
 ```
-pytest unidist/test/
+
+Third, using [config API](https://unidist.readthedocs.io/en/latest/flow/unidist/config.html) directly in your script:
+
+```python
+import unidist.config as cfg
+cfg.Backend.put("mpi") # unidist will use MPI backend to distribute computations
+import unidist.config as cfg
+cfg.Backend.put("ray") # unidist will use Ray backend to distribute computations
 ```
 
-For unidist MPI backend follow additional [instructions](unidist/core/backends/mpi/README.md).
+Default execution backend for unidist is Ray.
+
+#### Usage
+
+unidist provides [CLI interface](https://unidist.readthedocs.io/en/latest/using_cli.html) to run python programs.
+
+```python
+import unidist
+unidist.init() # Ray backend is used by default
+
+@unidist.remote
+def foo(x):
+    return x * x
+
+# This will run `foo` in parallel in workers;
+# `refs` will contain object references to actual data
```

Comment:
```suggestion
# Transform a `foo` in a unidist task and schedule it for parallel execution on a pool of workers;
# `refs` will contain object references to actual data
```

Reply:
Transforming  `foo` in a unidist task happens earlier. Also, I think this is a bit cumbersome phrase. Left it as is.

Reply:
Ok, but I can`t understand the "run ... parallel in workers" phrase. At least "in" is not correct here. *run .. on a pool of workers in parallel" better for me.

Reply:
fixed